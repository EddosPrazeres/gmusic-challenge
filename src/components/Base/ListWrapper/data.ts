export interface IListWrapper { 
  link?: string;
  ListElements: any;
  ItemComponent: any;
}

export interface IListWrapperLayout extends IListWrapper {  
  link?: string;
}