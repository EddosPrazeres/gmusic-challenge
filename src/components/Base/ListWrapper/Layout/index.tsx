import * as React from "react";
import { IListWrapperLayout } from "../data";
import './styles.scss'

export function ListWrapper({ ListElements, link, ItemComponent }: IListWrapperLayout) {
  return (
    <div className="ListWrapper"> 
      {ListElements.map((item: any) => (
        <ItemComponent {...item} link={link} />
      ))}
    </div>
  )
}