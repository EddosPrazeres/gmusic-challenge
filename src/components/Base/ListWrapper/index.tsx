import * as React from 'react';
import { IListWrapper } from './data';
import { ListWrapper as Layout } from './Layout';

export default function ListWrapper(props: IListWrapper) {
  return (
    <Layout {...props}/>
  )
}

  