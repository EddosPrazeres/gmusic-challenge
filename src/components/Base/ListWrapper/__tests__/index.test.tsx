import React from 'react'
import { mount, shallow } from 'enzyme'
import ListWrapper from '../index'

const createComponent = (props, useDefault = false) => {
  const componentProps = {    
    ...props,
  }

  const wrapper = useDefault ? mount(<ListWrapper  {...componentProps} />) : shallow(<ListWrapper     {...componentProps} />)
  const unmount = () => wrapper.unmount()

  return [wrapper, unmount]
}

describe('ListWrapper', () => {
  it('should match snapshot', () => {
    const [wrapper, unmount] = createComponent({})

    expect(wrapper).toMatchSnapshot()
    unmount()
  })
})
