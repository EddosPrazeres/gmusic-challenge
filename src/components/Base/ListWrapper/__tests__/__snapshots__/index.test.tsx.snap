// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListWrapper should match snapshot 1`] = `ShallowWrapper {}`;
